### ISSUE TITLE
e.Data.GetDataPresent not working as expected

### ISSUE THREAD

**Issue by anewton** [dotnet/docs-desktop#1415]
### Type of issue

Other (describe below)

### Description

In the instructions for section [To process the dropped data](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/walkthrough-enabling-drag-and-drop-on-a-user-control?view=netframeworkdesktop-4.8#to-process-the-dropped-data), the method e.Data.GetDataPresent was not working for a WPF app that targets net6.0-windows.  Replacing the code
`if (e.Data.GetDataPresent(DataFormats.StringFormat))`
with
`if (e.Data.GetData(DataFormats.StringFormat) != null)`
works.

### Page URL

https://learn.microsoft.com/en-us/dotnet/desktop/wpf/advanced/walkthrough-enabling-drag-and-drop-on-a-user-control?view=netframeworkdesktop-4.8

### Content source URL

https://github.com/dotnet/docs-desktop/blob/main/dotnet-desktop-guide/framework/wpf/advanced/walkthrough-enabling-drag-and-drop-on-a-user-control.md

### Document Version Independent Id

10411c7d-3586-4904-6d4f-79e84f0c9b98

### Article author

@adegeo

---
[Associated WorkItem - 292339](https://dev.azure.com/msft-skilling/Content/_workitems/edit/292339)

**Comment by adegeo** [comment]
Thank you for reporting this. I'll investigate and get back to you.

**Comment by adegeo** [comment]
Apologies @anewton though I'm sure you've moved on by now! I saw another feedback item come in related to this and it got me searching through the repository, and I rediscovered this one. Anyway, thanks again for the feedback and sorry this one got lost.

**Comment by adegeo** [comment]
User feedback from UUF:

> in the example code of "To process the dropped data" sectionï¼it should be "if (e.Data.GetDataPresent(DataFormats.Text))", not "if (e.Data.GetDataPresent(DataFormats.StringFormat))", otherwise the drop operation doesn't change the colour of the circle successfully.

**Comment by anewton** [comment]
No problem @adegeo.  I posted this for review and eventual fixing.  I'm not using WPF for this at the moment but others may find this page and would benefit from an update to the docs.  Thanks!

**Comment by adegeo** [comment]
Finally got to looking at this. I do remember there being various problems with drag-and-drop in .NET 6 and the older versions. I just tested the .NET Framework code documented in this article with .NET 8 and it works as intended.

Actually.. searching through the WPF issues I found this PR which fixes the problem. It was shipped a year ago. I even commented on it looking to see if it would be backported to .NET 6 but it wasn't

https://github.com/dotnet/wpf/pull/7028

I'm going to close this issue as there isn't anything to do here except tell customers to upgrade to the latest supported .NET version. I'll look into getting this documented as a breaking change in the .NET 6 breaking changes docs.
